presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -713,6 +714,7 @@ protected RelationPlan visitValues(Values node, Void context)
             rowsBuilder.add(values.build());
         }
 
+        context.incrementPlanNodes(session);
```

Comment:
do we have the same problem with values nodes as with table scans? I know it has the same effect on the width of the tree.

Reply:
Oh, the naming is a bit confusing as I also call this function in the `visitValues` function when creating Values node. Hence I just changed all related naming from \*ScanNode\* to \*LeafNode\*.

Reply:
We should also include IndexSourceNode